### PR TITLE
VXFM-10088 Adding SDRs to Flex OS Gateway Inventory

### PR DESCRIPTION
**Pull request by aimers1975**

### CODE REVIEW

File: bin/discovery.rb [hunk 1]
```diff
@@ -188,6 +189,14 @@ def protection_domain_sdslist(protection_domain)
   transport.post_request(url, {}, "get") || []
 end
 
+def protection_domain_sdrlist(protection_domain)
+  sp_url = "/api/instances/ProtectionDomain::%s/relationships/Sdr" % [protection_domain["id"]]
+  url = transport.get_url(sp_url)
+  transport.post_request(url, {}, "get") || []
+rescue
+  []
```
Comment (gavin-scott) [inline bin/discovery.rb]: Maybe a `logger.warn` would be good here? Is there a known case where it blows up? I'd rather just eat the errors in the known case and either raise or at least log the unknown  cases.

Comment (aimers1975) [reply bin/discovery.rb]: @gavin-scott yes can add a logger warn... any version of flex os earlier than 3.5, I realized too I should have called is sdr_url

File: bin/discovery.rb [hunk 2]
```diff
@@ -74,6 +74,7 @@ def collect_scaleio_facts
   end
 
   scaleio_system = scaleio_systems[0]
+  include_sdrs = scaleio_system["links"].find{ |link| link["href"].include?("Sdr")}
```
Comment (gavin-scott) [inline bin/discovery.rb]: Love it, nice find!

Comment (aimers1975) [reply bin/discovery.rb]: Uh, I think the suggestion was yours @gavin-scott  ;)

**Pull request merged**